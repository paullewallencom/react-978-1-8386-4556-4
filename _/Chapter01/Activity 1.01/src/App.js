import React, { Component } from "react";

const Header = props => <h1>{props.title}</h1>;

const InventoryItem = props => (
  <div className="InventoryItem">
    <strong>{props.itemName}</strong>
    <hr />
    <p>{props.itemPrice}</p>
  </div>
);

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      items: [
        { itemName: "Shoe", itemPrice: 5 },
        { itemName: "Sock", itemPrice: 3 }
      ]
    };
  }
  render() {
    return (
      <div className="App">
        <Header title="BuyStuff" />
        <InventoryItem
          itemName={this.state.items[0].itemName}
          itemPrice={this.state.items[0].itemPrice}
        />
        <InventoryItem
          itemName={this.state.items[1].itemName}
          itemPrice={this.state.items[1].itemPrice}
        />
      </div>
    );
  }
}

export default App;
