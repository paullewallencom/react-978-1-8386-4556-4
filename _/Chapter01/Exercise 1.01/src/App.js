import React from "react";

class App extends React.Component {
  render() {
    return (
      <div className="Example" id="my-element">
        <div
          className="greeting"
          style={{ background: "black", color: "white" }}
          onClick={() => alert("hello")}
        >
          Hello World
        </div>
      </div>
    );
  }
}

export default App;
