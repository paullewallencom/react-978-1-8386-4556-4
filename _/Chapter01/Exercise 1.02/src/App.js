import React from "react";

const App = props => (
  <div>
    <h1>My App</h1>
    <br />
    Hello React!
  </div>
);

export default App;
