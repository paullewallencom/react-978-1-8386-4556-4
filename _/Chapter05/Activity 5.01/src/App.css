.container {
  max-width: 960px;
  margin: 0 auto;
  padding: 0 10%;
}

.center {
  text-align: center;
}
