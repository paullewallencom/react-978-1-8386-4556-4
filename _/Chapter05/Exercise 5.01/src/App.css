.App {
  text-align: left;
  margin: 50px;
}
