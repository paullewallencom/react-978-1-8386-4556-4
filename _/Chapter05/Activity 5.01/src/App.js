import React from "react";
// styles
import "./App.css";
// components
import CommentSection from "./components/CommentSection/";
// data
import { comments } from "./comments.json";

function App() {
  return (
    <div className="container">
      <CommentSection comments={comments} />
    </div>
  );
}

export default App;
