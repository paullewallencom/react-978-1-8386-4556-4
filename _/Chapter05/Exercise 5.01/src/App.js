import React from "react";
import "./App.css";
import Profile from "./Profile";

const user = {
  name: "Brian",
  interests: "Reading, Swimming, Technology",
  age: 9,
  image:
    "https://images.unsplash.com/photo-1470071131384-001b85755536?auto=format&fit=crop&w=200&q=80",
  color: "Red",
  movie: "Star Wars"
};

function App() {
  return (
    <div className="App">
      <Profile user={user} />
    </div>
  );
}

export default App;
