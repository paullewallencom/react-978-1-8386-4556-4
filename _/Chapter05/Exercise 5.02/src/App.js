import React from "react";
import "./App.css";
import Loader from "./Loader";

function App() {
  return (
    <div className="App">
      <Loader spinnerSize={60} isVisible={true} />,
    </div>
  );
}

export default App;
