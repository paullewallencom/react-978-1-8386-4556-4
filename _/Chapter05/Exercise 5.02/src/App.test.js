import React from "react";
import { render, cleanup } from "@testing-library/react";
import App from "./App";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

test("renders div with class App", () => {
  const { container } = render(<App />);
  expect(container.firstChild.classList.contains("App")).toBe(true);
});

test("renders component with class spinner", () => {
  const { container } = render(<App />);
  expect(container.firstChild.firstChild.classList.contains("spinner")).toBe(
    true
  );
});
