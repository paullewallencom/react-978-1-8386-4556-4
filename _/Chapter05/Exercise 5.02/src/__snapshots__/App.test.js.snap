// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="spinner"
    >
      <svg
        focusable="false"
        height="60"
        viewBox="0 0 60 60"
        width="60"
      >
        <circle
          cx="30"
          cy="30"
          r="20"
        />
      </svg>
    </div>
    ,
  </div>
</DocumentFragment>
`;
